torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -399,7 +399,10 @@ at::Tensor& AtenXlaType::addcmul_(at::Tensor& self, const at::Tensor& tensor1,
 at::Tensor AtenXlaType::addmm(const at::Tensor& self, const at::Tensor& mat1,
                               const at::Tensor& mat2, at::Scalar beta,
                               at::Scalar alpha) {
-  if (beta.to<double>() != 1 || alpha.to<double>() != 1) {
+  // xla::dot doesn't support integer types.
+  if (beta.to<double>() != 1 || alpha.to<double>() != 1 ||
+      !self.is_floating_point() || !mat1.is_floating_point() ||
```

Comment:
Does pytorch mm support multiplying floating and integers?
If not, we could simplify the test maybe?

Reply:
pytorch doesn't allow multiply floating and integers. 
`RuntimeError: Expected object of scalar type Float but got scalar type Long for argument #2 'vec' in call to _th_mv`
Btw the tests are not for different types, it's for different shapes are sent to different calls in `matmul`. 
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/LinearAlgebra.cpp#L348-L443
So I think it's somehow necessary to test all of them :(